### PR TITLE
Add action background

### PR DESCRIPTION
**Pull request by strausr**
#### What does this PR solve?
Adds action background

#### Final checklist
- [x] JSdoc - Add proper docstrings based on our PHP implementation
- [x] JSdoc - Hide private functions using @private
- [ ] Implementation is aligned with PHP Reference(If different, please specify why)
- [x] Tests - Add proper tests to the added code

### CODE REVIEW

File: src/actions/background/Background.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Defines the background color to use instead of transparent background areas or when resizing with padding.
+ *
+ * **Learn more**:
+ * <a href="https://cloudinary.com/documentation/image_transformations#setting_background_color" target="_blank">
+ * Setting background for images</a> |
+ * <a href="https://cloudinary.com/documentation/video_manipulation_and_delivery#background_color" target="_blank">
+ * Setting background for videos</a>
+ *
+ * @api
```
Comment (patrick-tolosa) [inline src/actions/background/Background.ts:10]: Unfortunately, we can't just copy the docstrings from PHP, certain things need to change.

Also, please make sure this looks proper when you build the docs (npm run build:docs).
Pay attention to line breaks and spaces.

Also, `Background` should be part of the Actions namespace, when building the docs ensure there's an `Actions.Background` entry (under namespaces)

File: src/actions/background/IBackgroundAction.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import {IAction} from "../../interfaces/IAction";
+
+export interface IBackgroundAction extends IAction{
```
Comment (patrick-tolosa) [inline src/actions/background/IBackgroundAction.ts:3]: I know we used these interfaces before, but I'd like to question if we really need them.
Can this be removed?

File: src/actions/background/auto.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import BackgroundAction from "./BackgroundAction";
+import {IBackgroundAction} from "./IBackgroundAction";
+
+/**
+ * Applies background color automatically.
+ */
+function auto(): IBackgroundAction{
```
Comment (patrick-tolosa) [inline src/actions/background/auto.ts:7]: Functions without a namespace applied to them (in the docstring) will appear as public in the JSdoc,
This may or may not be what you wanted to implement.

File: src/actions/resize/IResizeAction.ts [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import {IAction} from "../../interfaces/IAction";
 import {GravityParam} from "../../params/gravity/Gravity";
+import {IBackgroundAction} from "../background/IBackgroundAction";
 
 export interface IResizeAction extends IAction{
```
Comment (patrick-tolosa) [inline src/actions/resize/IResizeAction.ts:5]: The IResizeAction is scheduled for removal in a different PR, can you please undo the changes here?

**Review by patrick-tolosa — CHANGES_REQUESTED** [review]
Needs adjustments for JSdoc strings to ensure things are placed properly.

File: src/actions/background/BackgroundAction.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import Param from "../../parameters/Param";
+import ParamValue from "../../parameters/ParamValue";
+
+class BackgroundAction extends Param implements IAction {
+  constructor(backgroundType?: string) {
+    super('b', new ParamValue(backgroundType).setDelimiter('_'));
+  }
+
+  /**
+   * @memberOf Actions.Background
```
Comment (patrick-tolosa) [inline src/actions/background/BackgroundAction.ts:11]: This method is not actually a member of Actions.Background right?

A user cannot import {contract} from actions/background.

**Review by patrick-tolosa — APPROVED** [review]

File: src/actions/background/BackgroundAction.ts (hunk 5 above)
Comment (strausr) [reply src/actions/background/BackgroundAction.ts:11]: But it is public. How should we add it to the docs?
